docs/production/authentication-methods.md

```diff
@@ -10,186 +10,6 @@ chosen authentication methods require; then restart the Zulip server.
 
 Details on each method below.
 
-## Email and password
```

Comment:
I don't like the ordering change in this file -- "Email and Password" and the SSO are simpler, and should appear before the super long-and-complex LDAP and SAML.

Reply:
Move "Email and Password" and "Plug-and-play SSO" section before LDAP.

Reply:
Can you rebase to fix the errors in the commits that introduced them, rather than having an extra fixup commit later?